src/exo/LoopIR_unification.py

```diff
@@ -1120,19 +1142,23 @@ def unify_e(self, pe, be):
         elif isinstance(pe, LoopIR.USub):
             self.unify_e(pe.arg, be.arg)
         elif isinstance(pe, LoopIR.BinOp):
+            exprs = [pe.rhs, pe.lhs, be.rhs, be.lhs]
+            if pe.op in comparision_ops and all(e.type.is_indexable() for e in exprs):
+                inequality_ops = comparision_ops - {"=="}
+                if pe.op == be.op or (
+                    pe.op in inequality_ops and be.op in inequality_ops
+                ):
```

Comment:
I think this branch is taken when unifying index equality operations like `if a + 1 == b : ...` and `if n == m: ...`, and it seems you're not handling the unification of LHS

Reply:
Those would go through the first case in `comparision_to_unification_expr`. They will be rewritten into `0 == b - (a + 1)` and `0 == m - n`, then we will unify `b-(a+1)` and `m-n`.

Reply:
Ah gocha